transport/http/inbound_test.go

```diff
@@ -37,6 +40,32 @@ import (
 	"golang.org/x/net/context"
 )
 
+func TestStartAddrInUse(t *testing.T) {
+	i1 := NewInbound(":0")
+	require.NoError(t, i1.Start(new(transporttest.MockHandler)))
+	i2 := NewInbound(i1.Addr().String())
+	err := i2.Start(new(transporttest.MockHandler))
+
+	require.Error(t, err)
+	oe, ok := err.(*net.OpError)
+	assert.True(t, ok && oe.Op == "listen", "expected a listen error")
+	if ok {
+		se, ok := oe.Err.(*os.SyscallError)
+		assert.True(t, ok && se.Syscall == "bind" && se.Err == syscall.EADDRINUSE, "expected a EADDRINUSE bind error")
+	}
+
+	assert.NoError(t, i1.Stop())
+}
+
+func TestNilAddrAfterStop(t *testing.T) {
+	i := NewInbound(":0")
+	require.NoError(t, i.Start(new(transporttest.MockHandler)))
+	assert.NotEqual(t, ":0", i.Addr().String())
+	assert.NotNil(t, i.Addr())
+	assert.NoError(t, i.Stop())
+	assert.Nil(t, i.Addr())
+}
+
```

Comment:
you can have a test where you close the listener directly, then close again using the `Stop` method. You should get an error, and the test will also determine the priority between errors